2025/day2.md

```diff
@@ -1,46 +1,200 @@
-# Day 2 – [Lab Title Here]
+# Day 2 – Linux Kernel Modules: Character Drivers, GPIO, and User-Space Communication
 
 Slides: [day2_slides](./day2_slides)
 
 ## Description
 
-_Provide a short description of what this lab session is about. Mention the scope and practical goals
-for the day, what students are expected to understand or achieve by the end of this lab._
+This lab introduces key concepts in Linux kernel module development, with a focus on character drivers and the interaction between user space and kernel space. Through a sequence of practical exercises, the lab explores the use of file operations, ioctl, and GPIO-based hardware control. The final goal is to build a solid understanding of how user applications can safely communicate with kernel modules to perform data exchange and hardware manipulation.
```

Comment:
character device drivers

ioctl is a file operation, no need to mention it explicitly